src/services/directory.ts

```diff
@@ -4,7 +4,47 @@ import { devPrint } from '../components/utils/RandomUtils';
 import { AxiosResponse } from 'axios';
 import { deserializeMember } from './member';
 
-export async function searchMembers(nameQuery: string): Promise<Member[]> {
+interface Cache<T> {
```

Comment:
Do we want to put this in it's own file and use it for other places? I'm not sure where the cache would make sense to add elsewhere but might be useful?

Reply:
yeah we can, but I'll do this in a separate PR. Directory is a particularly good candidate for this bc it is read only and we don't need consistent views of other profiles, but i'm sure it'll be useful for other endpoints too